tests/test_loan_transitions.py

```diff
@@ -545,26 +535,11 @@ def test_document_requests_on_item_returned(
             assert pending_loan["document_pid"] == "document_pid"
 
 
-def test_loan_deny_item_request(
-        loan_created, db, params, mock_can_not_be_requested):
-    """Test loan item request action."""
-
-    with pytest.raises(RecordCannotBeRequested):
-        current_circulation.circulation.trigger(
-            loan_created,
-            **dict(
-                params,
-                trigger="request",
-                pickup_location_pid="pickup_location_pid",
-            )
-        )
-
-
-def test_loan_deny_document_request(
-        loan_created, db, params, mock_can_not_be_requested):
-    """Test loan document request action."""
-    del params['item_pid']
-    params['document_pid'] = 'document_pid'
+def test_deny_request(loan_created, db, params):
+    """Test deny request action."""
+    current_app.config["CIRCULATION_POLICIES"]["request"][
```

Comment:
So, here is better to use the `SwappedNestedConfig` https://github.com/inveniosoftware/invenio-circulation/blob/e3e637afd1c0a7aa1e994ae71adea02c8cba842c/tests/test_loan_transitions.py#L221. I am saying that because you override here the policy so in the future another test that comes after this one will fail as the policy would return always false. The purpose of the `SwappedNestedConfig` is to change a config only locally for the test purposes and then change it back to what it was set before :)

Reply:
Changed